pkg/formatters/event.go

```diff
@@ -40,9 +42,28 @@ func ResponseToEventMsgs(name string, rsp *gnmi.SubscribeResponse, meta map[stri
 		return nil, nil
 	}
 	evs := make([]*EventMsg, 0, len(rsp.GetUpdate().GetUpdate())+len(rsp.GetUpdate().GetDelete()))
+	response := rsp
 	switch rsp := rsp.Response.(type) {
 	case *gnmi.SubscribeResponse_Update:
 		namePrefix, prefixTags := tagsFromGNMIPath(rsp.Update.GetPrefix())
+		// Extension message to tags
+		if prefixTags == nil {
+			prefixTags = make(map[string]string)
+		}
+		for _, ext := range response.Extension {
+			extensionValues, err := extensionToMap(ext)
+			if err != nil {
+				return nil, err
+			}
+			for k, v := range extensionValues {
+				switch v := v.(type) {
+				case string:
+					prefixTags[k] = v
+				case float64:
+					prefixTags[k] = strconv.FormatFloat(v, 'G', -1, 64)
+				}
+			}
+		}
```

Comment:
This part assumes that any extension data can and will be mapped to a map[string]any, and that the goal is to get the extension data as tags, it will not always be the case for all extensions.